pkg/apis/autoscaling/v1/override_types.go

```diff
@@ -120,6 +121,20 @@ type PodResourceOverride struct {
 	Spec            PodResourceOverrideSpec `json:"spec,omitempty"`
 }
 
+// DeploymentOverrides defines fields that can be overridden for a given deployment.
+type DeploymentOverrides struct {
+	// Override the NodeSelector for the deployment's pods. This allows, for example, for the ClusterResourceOverride
+	// to be run on non-master nodes.
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+
+	// Override the Tolerations of the deployment's pods. This allows, for example, for the ClusterResourceOverride
+	// to be run on non-master nodes with a specific taint.
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+
+	// Override the number of replicas for the deployment.
+	Replicas *int32 `json:"replicas,omitempty"`
```

Comment:
I know in VPA we tried to stay close to the `DeploymentSpec` types but I think I agree with you that this makes more sense as a pointer here for now, and I don't think it should hurt us long-term.  (Otherwise we'd have to mark the zero value as ignored, or backfill this with the existing value if a DeploymentOverride was supplied, which ughhhh seems like a pain)

Reply:
Sounds good, thanks!